aws/src/test/java/org/apache/iceberg/aws/s3/TestS3FileIO.java

```diff
@@ -218,8 +219,7 @@ public void testPrefixList() {
   public void testPrefixDelete() {
     String prefix = "s3://bucket/path/to/delete";
     List<Integer> scaleSizes = Lists.newArrayList(0, 5, 1000, 2500);
-
-    scaleSizes.parallelStream().forEach(scale -> {
+    scaleSizes.forEach(scale -> {
```

Comment:
The original rational for using the parallel stream was to help speedup the tests because it take the S3 mock library a considerable amount of time to create 2.5K files (even all locally).  The intent of the scale factors was to ensure that pagination in the s3 list calls is working correctly.  If these tests are run in parallel with the other project tests, I think it's probably fine to just call sequentially.

Reply:
Yeah it makes sense to parallelize because S3 mock still takes a lot of time in the high scale case. The unit tests are run concurrently, so I think we're good there. 

Since the issue seems to lie with S3 mock, I think to truly validate this behavior it makes sense to have integ tests which validate against S3 but for now I think it makes sense to just address in the unit tests. Thanks!

Reply:
In that case can we reduce the number of files created for local test, and create another integration test for the same use case with parallel delete?

Reply:
Makes sense, the pagination is handled by the AWS SDK listObjectsV2Paginator so effectively that's already unit tested through the AWS SDK. Integ tests can validate the actual behavior and we can do a parallel delete as you mentioned. Updated and ran integ tests, which are passing.